src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -947,8 +947,16 @@ private void Open(object parameters)
             // that can't be handled reliably
             try
             {
-                string xmlFilePath = parameters as string;
+                var xmlFilePath = parameters as string;
                 ExecuteCommand(new DynamoModel.OpenFileCommand(xmlFilePath));
+
+                if (UIDispatcher != null)
+                {
+                    // Dispatch a fit view command after all non-idle operations
+                    // are completed. This is to ensure the view's ready for it.
+                    UIDispatcher.BeginInvoke(DispatcherPriority.Background,
+                        (Action)(() => { if (CanFitView(null)) FitView(null); }));
+                }
             }
```

Comment:
All of this code related to the `Dispatcher` was setting off alarms in my head.  The question is - is it necessary?  PropertyChanged events are automatically marshalled to the UI thread, and it would seem that `FitView` would only do that, but maybe I'm wrong.  Either way, I'm a bit weary looking at this code.